components/bootloader_support/src/esp_image_format.c

```diff
@@ -225,7 +243,7 @@ static esp_err_t image_load(esp_image_load_mode_t mode, const esp_partition_pos_
     }
 
 #ifdef BOOTLOADER_BUILD
-    if (do_load) { // Need to deobfuscate RAM
+    if (ram_obfs_value[0] != 0 && ram_obfs_value[1] != 0 && do_load) { // Need to deobfuscate RAM
```

Comment:
Minor, but can this be `(do_load && (ram_obfs_value[0] != 0 || ram_obfs_value[1] != 0))`?

One of the two ram_obfs_values will be zero once in every 2^31 boots, but both will only be zero once in every 2^64 boots. I like the second odds better. :)

Reply:
(Or test the `do_verify` flag as suggested in the other comment, maybe? As ram_obfs_value being zero is being tested because it's a side effect of skipping verification, yes?)

Reply:
> Minor, but can this be `(do_load && (ram_obfs_value[0] != 0 || ram_obfs_value[1] != 0))`?
> 
> One of the two ram_obfs_values will be zero once in every 2^31 boots, but both will only be zero once in every 2^64 boots. I like the second odds better. :)

It's guaranteed to be non-zero on a normal boot - take a look a the existing code that generates ram_obfs_value[x].  It has the following loop:

```
    while (ram_obfs_value[0] == 0 || ram_obfs_value[1] == 0) {
        ....
    }
```

Reply:
Oh, my mistake! Thanks.